### PR TITLE
[fix](case function) Applying for an array in the heap to avoid stack overflow

### PR DESCRIPTION
**Pull request by zxealous**
… overflow

## Proposed changes

Issue Number: close #xxx

Applying for an array in the heap to avoid stack overflow.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/functions/function_case.h [hunk 1]
```diff
@@ -167,9 +167,9 @@ class FunctionCase : public IFunction {
         int rows_count = column_holder.rows_count;
 
         // `then` data index corresponding to each row of results, 0 represents `else`.
-        int then_idx[rows_count];
-        int* __restrict then_idx_ptr = then_idx;
-        memset(then_idx_ptr, 0, sizeof(then_idx));
+        std::unique_ptr<int> then_idx(new int[rows_count]);
```
Comment (yiguolei) [inline be/src/vec/functions/function_case.h:170]: the rows_count maybe too large? I think it should be less than 4096, do you really meet this problem?

Comment (zxealous) [reply be/src/vec/functions/function_case.h:170]: I have meet this problem in 1.1,  the rows_count is  > 5000w, cause all backends core dump, the problem was fixed by #17035, but it is hard to pick it to branch 1.1. I think changing it on the master can also avoid the same problem again.

**Comment by zxealous** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 50.59 seconds
 stream load tsv:          507 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          28.8 seconds inserted 10000000 Rows, about 347K ops/s
 storage size: 17161383030 Bytes

**Comment by zxealous** [comment]
run p0

**Comment by zxealous** [comment]
run buildall

**Comment by zxealous** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 50.74 seconds
 stream load tsv:          507 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.1 seconds inserted 10000000 Rows, about 343K ops/s
 storage size: 17160797810 Bytes

File: be/src/vec/functions/function_case.h (hunk 1 above)
Comment (zxealous) [reply be/src/vec/functions/function_case.h]: @yiguolei plz help me cc

**Pull request closed without merging**